### PR TITLE
[SPARK-44990][SQL][FOLLOWUP] Remove lazy of `nullAsQuotedEmptyString`

### PR DESCRIPTION
**Pull request by Hisoka-X**
### What changes were proposed in this pull request?
This is a follow up PR of #42738 . Remove lazy of `nullAsQuotedEmptyString`.

### Why are the changes needed?
Remove lazy of nullAsQuotedEmptyString. Because most case it useless.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
add new test.

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityGenerator.scala [hunk 1]
```diff
@@ -60,7 +60,7 @@ class UnivocityGenerator(
     options.locale,
     legacyFormat = FAST_DATE_FORMAT,
     isParsing = false)
-  private lazy val nullAsQuotedEmptyString =
+  private val nullAsQuotedEmptyString =
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityGenerator.scala:63]: I don't think there'd be meaningful result in the benchmark :-). LGTM

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVBenchmark.scala [hunk 2]
```diff
@@ -378,6 +394,7 @@ object CSVBenchmark extends SqlBasedBenchmark {
       countBenchmark(rowsNum = 10 * 1000 * 1000, numIters)
       datetimeBenchmark(rowsNum = 10 * 1000 * 1000, numIters)
       filtersPushdownBenchmark(rowsNum = 100 * 1000, numIters)
+      writeNullBenchmark(rowsNum = 10 * 1000 * 1000, numIters)
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVBenchmark.scala:397]: it isn't worth adding benchmark for writing all null values...

Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVBenchmark.scala:397]: it's not a meaningful benchmark, I'd like to revert it

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**